### ISSUE TITLE
5 second timeout causes failures for larger Hydrus libraries

### ISSUE THREAD

**Issue by floogulinc** [ififfy/flipflip#261]
The 5 second timeout appears to not be just for individual requests but the whole process of getting the list and metadata from Hydrus, this will fail on even somewhat large libraries as all the metadata requests will take a while. Hydrus doesn't handle a ton of simultaneous requests very well either so it would probably be good to stagger them instead of doing them all in parallel or you will see some of the metadata requests error.

**Comment by professor-mesmerize** [comment]
@floogulinc Just to clarify, in terms of staggering requests, do you mean specifically staggering the chunked `/file_metadata` requests that get made if `/search_files` returns more than 1000 ids?

**Comment by floogulinc** [comment]
@professor-mesmerize Yes, right now these metadata requests all get made in parallel. In Hydrus Web I have it limited to 8 requests at a time for metadata in the comics finder system which is similar to this in terms of doing a large search and getting metadata for a large number of files at once.

**Comment by StarrHelixx** [comment]
[This chunk](https://github.com/ififfy/flipflip/blob/master/src/renderer/components/player/SourceScraper.tsx#L1524-L1527) could be improved to create a delay between metadata request. And/or the timeouts for these request could be increased from 5000ms

**Comment by professor-mesmerize** [comment]
@StarrHelixx I can probably put up a PR for this one as well! That said, I'm curious about what I'm seeing in the network calls for Hydrus - namely, I'm curious about why FlipFlip regularly makes the same `/file_metadata` request with the exact same fileIDs, over and over, while playing from a Hydrus source. Does it not make sense in some way to only fetch more file URLs (via calling `/search_files` and `/file_metadata`) when the number of stored Hydrus file URLs is getting low? Or are all remote sources configured to be re-checked every few seconds?

**Comment by StarrHelixx** [comment]
@floogulinc Ok I think this one is solved now. See https://github.com/ififfy/flipflip/commit/0d484396cf743a0eb5ac5a903886c81ec73a399a .  I added code to stagger all metadata requests, so there will only be 1 per second. Is this sufficiently slow? I could also make it so it only ever did one at a time, but I figured this might allow for a few simultaneous requests without overloading Hydrus.

> I'm curious about why FlipFlip regularly makes the same /file_metadata request with the exact same fileIDs, over and over, while playing from a Hydrus source.
@professor-mesmerize FlipFlip will keep pulling images, even after it has already loaded them. This is to keep the slideshow moving, without having to keep _all_ previously played images in memory. In fact, Chromium _should_ do a decent job of caching those image calls, and serving the already loaded image if it's available (but I'm not 100% if that's exactly what happens)
